app/models/spree/line_item_decorator.rb

```diff
@@ -135,8 +141,7 @@ def stock_availability
   end
 
   def scoper
-	  return @scoper unless @scoper.nil?
-	  @scoper = OpenFoodNetwork::ScopeVariantToHub.new(order.distributor)
+    @scoper ||= OpenFoodNetwork::ScopeVariantToHub.new(order.distributor)
```

Comment:
This could be quite inefficient. The scoper loads all variant overrides of an enterprise. Are we doing this for only one line item or for a long list of line items? Load time would grow with `#items * #overrides`.

Reply:
A quick fix would be to load the override for this one line item and supply that to the scoper.

Reply:
I don't think that's the scope of this PR or #3147 
This is the way of loading variant overrides in master right now: you load all of them once.
Do you think this needs to be changed? I'd open a separate issue for that.
Or am I missing something?

Reply:
I think we always load variant overrides for a collection in master, all at once. Here you introduce a method that loads variant overrides individually, but you use the scoper that was designed for batch processing and loads all variant overrides. This pull request is introducing a performance issue by using the scoper differently to before. The quick fix would be only one additional line. It's not very efficient when doing that with a lot of variants, but much more efficient than the current code.

```suggestion
    @scoper ||= OpenFoodNetwork::ScopeVariantToHub.new(
      order.distributor,
      Hash[variant, VariantOverride.for(order.distributor, variant)]
    )
```

Reply:
I don't get it :-)
I copied this code from here:
https://github.com/openfoodfoundation/openfoodnetwork/blob/66585e93a85f1d557a54345d52af1bac822f3a1f/app/models/spree/line_item_decorator.rb#L165-L168

and because variant_overrides is nil this will load all variants of the hub:
https://github.com/openfoodfoundation/openfoodnetwork/blob/66585e93a85f1d557a54345d52af1bac822f3a1f/lib/open_food_network/scope_variant_to_hub.rb#L3-L6

using the vo.indexed scope: https://github.com/openfoodfoundation/openfoodnetwork/blob/66585e93a85f1d557a54345d52af1bac822f3a1f/app/models/variant_override.rb#L30-L33

Reply:
Ah, thank you! You are right that this a pre-existing problem, because it was used in the line item decorator already.

I still think it's very inefficient to load all variant overrides of a hub (potentially thousands) and then use only one of them. If that happens in a loop, we burn a lot of energy on nothing. But you are right, it's a pre-existing problem.